sympy/polys/domains/domain.py

```diff
@@ -864,8 +864,15 @@ def poly_ring(self, *symbols, order=lex):
 
     def frac_field(self, *symbols, order=lex):
         """Returns a fraction field, i.e. `K(X)`. """
+        from sympy.core.power import Pow
         from sympy.polys.domains.fractionfield import FractionField
-        return FractionField(self, symbols, order)
+        _symbols = []
+        for symbol in symbols:
+            if isinstance(symbol, Pow) and symbol.exp.is_number and symbol.exp < 0:
+                _symbols.append(1/symbol)
```

Comment:
This check would also pass if the exponent is something like `-sqrt(2)`. The check should be `is_Integer` to be sure that it is an explicit negative integer. Also rather than `1/symbol` it is more direct to use `symbol.base`. So then it would be:
```
if isinstance(symbol, Pow) and symbol.exp.is_Integer and symbol.exp < 0:
    _symbols.append(symbol.base)
```
Then again if we're doing this for negative integer powers we could also do it for positive integer powers.

Reply:
yeah true ,I had given a thought about this some time back and I had realized that `is_number` won't do the designated job correctly , though the error lies in negative integer powers so maybe we could restrict it to the negative powers for now or untill we find any error !

Reply:
I have made changes in the check in the latest commit !